onboarding_auth/onboardingAuthPresentation/src/main/java/com/alexafit/onboardingAuthPresentation/login/LoginScreen.kt

```diff
@@ -0,0 +1,157 @@
+package com.alexafit.onboardingAuthPresentation.login
+
+import androidx.compose.foundation.background
+import androidx.compose.foundation.clickable
+import androidx.compose.foundation.layout.Arrangement
+import androidx.compose.foundation.layout.Box
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.Row
+import androidx.compose.foundation.layout.Spacer
+import androidx.compose.foundation.layout.fillMaxSize
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.height
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.shape.RoundedCornerShape
+import androidx.compose.material.MaterialTheme
+import androidx.compose.material.ScaffoldState
+import androidx.compose.material.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.LaunchedEffect
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.platform.LocalContext
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.text.input.ImeAction
+import androidx.compose.ui.text.style.TextAlign
+import androidx.hilt.navigation.compose.hiltViewModel
+import com.alexafit.core.util.UiEvent
+import com.alexafit.coreui.LightBlue
+import com.alexafit.coreui.LocalSpacing
+import com.alexafit.coreui.components.buttons.TextActionButton
+import com.alexafit.coreui.components.textfield.TextField
+import com.alexafit.onboardingAuthPresentation.R
+import com.alexafit.onboardingAuthPresentation.login.event.LoginEvent
+
+@Composable
+fun LoginScreen(
+    scaffoldState: ScaffoldState,
+    onEventClick: (LoginEvent) -> Unit,
+    viewModel: LoginViewModel = hiltViewModel()
+) {
+    val spacing = LocalSpacing.current
+    val context = LocalContext.current
+    LaunchedEffect(key1 = true) {
+        viewModel.uiEvent.collect { event ->
+            when (event) {
+                is UiEvent.Success -> onEventClick(LoginEvent.Login)
+                is UiEvent.ShowSnackbar -> {
+                    scaffoldState.snackbarHostState.showSnackbar(
+                        message = event.message.asString(context)
+                    )
+                }
+            }
+        }
+    }
+    Box(
+        modifier = Modifier
+            .fillMaxSize()
+            .background(MaterialTheme.colors.background)
+    ) {
```

Comment:
Are you using a `Surface` as root for all your screens? A surface would make sure to always adjust your text and icon colors based on the background. E.g. if your `MaterialTheme.colors.background` would resolve to a dark color here, Texts would still be black by default, even in dark theme (unless specified otherwise of course). A surface makes sure that the texts & icons always have the right content color based on the background

Reply:
I did not know a surface would resolve this, so I created my own personal identifier to determine what colors are presented in light and dark theme in my Theme.xml file in the app module

Reply:
![image](https://user-images.githubusercontent.com/64118964/227101184-cabe35f9-73c2-46e1-b2d9-1ed88983eca9.png)

Reply:
This way I can atleast programmatically determine which color will be presented in a light or dark theme

Reply:
That's correct, you created a theme. Let's assume the device is in dark mode. What I was meaning is this:
```
Box(
    modifier = Modifier
            .fillMaxSize()
            .background(MaterialTheme.colors.background)
) {
    Text("Hello world") // This text will be **black** on a dark background
}

Surface(
    modifier = Modifier.fillMaxSize(),
    color = MaterialTheme.colors.background
) {
    Text("Hello world") // This text will be **white** on a dark background because the surface automatically adjusts it
}
```